proposals/build-provenance-and-code-signing-for-homebrew.md

```diff
@@ -0,0 +1,112 @@
+# Build Provenance and Code-signing for Homebrew
+
+Authors: [William Woodruff](https://github.com/woodruffw), [Dustin Ingram](https://github.com/di)
+
+## Summary
+
+The document contains a technical proposal for introducing build provenance and cryptographic signatures to the Homebrew package manager. The end goal is to improve Homebrew's authenticity and supply-chain integrity properties, establishing that a given package was both built from an attested source artifact as well as on an attested machine (i.e., a Homebrew-controlled GitHub Actions workflow).
+
+The proposal is broken up into individual technical items, which are meant to be semi-independent units of work. The units of work are ordered by dependency relationship; i.e. signature generation must come before any signatures can be verified on the client side.
+
+## Technical Background
```

Comment:
Please add a threat model, because without one, it's hard for the unfamiliar reader to see exactly what this proposal solves for.

Is it fair to say that the threat model is that attackers who compromise the object storage (whatever that is, whether a GitHub repo, or an OCI registry) cannot pretend to be the authors of the package, at least w/o compromising GitHub Actions, Sigstore Fulcio, and/or the GitHub repo for the package?

Reply:
https://github.com/ossf/wg-securing-software-repos/pull/25

Reply:
> Is it fair to say that the threat model is that attackers who compromise the object storage (whatever that is, whether a GitHub repo, or an OCI registry) cannot pretend to be the authors of the package, at least w/o compromising GitHub Actions, Sigstore Fulcio, and/or the GitHub repo for the package?

Correct: this is scenario (1) in the PR I linked above.